File: xls-sql-converter/src/main/java/ru/yandex/market/v1/writer/DbDataWriter.java
package ru.yandex.market.v1.writer;

import java.io.IOException;
import java.io.UncheckedIOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

import ru.yandex.market.v1.table.Table;

import static java.nio.file.StandardOpenOption.CREATE;
import static java.nio.file.StandardOpenOption.TRUNCATE_EXISTING;

public abstract class DbDataWriter {

    public abstract void write(List<Table> tables, String target);

    protected void writeFile(List<String> lines, String target) {
        try {
            Files.write(Paths.get(target), lines, CREATE, TRUNCATE_EXISTING);
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
        System.out.println("File saved: " + target);
    }
}

File: xls-sql-converter2/src/main/kotlin/app/converter/writer/DbDataWriter.kt
package app.converter.writer

import app.converter.table.Table
import java.nio.file.Files
import java.nio.file.Paths
import java.nio.file.StandardOpenOption

abstract class DbDataWriter {
    abstract fun write(tables: List<Table>, target: String)

    protected fun writeFile(lines: List<String>, target: String) {
        Files.write(Paths.get(target), lines, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
        println("File saved: $target")
    }
}